File: Javascript/Authored/scooby.js
function scoobydoo(villian, villians) {
  villian = villian.toLowerCase();

  // Step 1 - Rotate Right by 5
  villian = rotate(villian.split(""),5, true).join("");

  // Step 2 - Reverse
  villian = villian.split("").reverse().join("");

  // Step 3 - Change even letters by 5
  villian = villian.split("");
  for (var i = 0; i < villian.length; i++) {
    if ((i+1) % 2 === 0){villian[i] = shift(villian[i],5);}
  }
  villian = villian.join("");

  for (i = 0; i < villians.length; i++) {
    if (villians[i].split(" ").join("").toLowerCase() === villian){return villians[i];}
  }
  return "";
}

File: Python/Authored/scooby.py
def scoobydoo(villian, villians):
    villian = villian.lower()

    # Step 1 - Rotate Right by 5
    villian = ''.join(shift(villian, 5))

    # Step 2 - Reverse
    villian = villian[::-1]

    # Step 3 - Change even letters by 5
    badguy = ""
    num = 0
    for le in villian:
        num += 1
        if num % 2 == 0:
            le = le.translate(str.maketrans('abcdefghijklmnopqrstuvwxyz','fghijklmnopqrstuvwxyzabcde'))
        badguy += le

    for bg in villians:
        if ''.join(bg.lower().split(' ')) == badguy: return bg
